suites/message/message_application.go

```diff
@@ -116,5 +120,11 @@ func TestMessageApplicationEdgecases(t *testing.T, factory state.Factories) {
 			}, chain.Nonce(1), chain.Value(big_spec.Zero())),
 			types.MessageReceipt{ExitCode: exitcode.ErrIllegalArgument, ReturnValue: drivers.EmptyReturnValue, GasUsed: abi_spec.NewTokenAmount(1_000_000)},
 		)
+
+		// assert that the entire gas limit is deducted
+		gasReward := big.NewInt(td.MessageProducer.DefaultGasLimit())
+		gasReward = gasReward.Mul(gasReward, td.MessageProducer.DefaultGasPrice().Int)
+		senderBalance.Sub(senderBalance.Int, gasReward)
+		td.AssertBalance(sender, senderBalance)
```

Comment:
Unsure this is the correct behaviour. The [ApplyMessage](https://filecoin-project.github.io/specs/#vminterpreter-implementation) section of the spec indicates that unused gas is to be refunded to the caller iff the sender address is valid (which it is here).

Reply:
This change is invalid. Only the actually consumed gas should be spent.

If you believe there is disagreement about this, please remove this change so that this PR can otherwise land, and open an issue for discussion in specs.